src/upnp.cpp

```diff
@@ -965,8 +965,31 @@ void upnp::on_upnp_xml(error_code const& e
 		return;
 	}
 
+	// determine IGD version
+	std::string igd_str(p.get_body().begin, p.get_body().end);
```

Comment:
looking at this code, doesn't it look a bit backwards? There is an xml parser below, so why just do a string-match search first? It seems more natural to parse out the version after the xml parser has found the key. I realize it's not very likely the xml document will have an occurrence of this string elsewhere, but the code implies that it is impossible, and it is slightly confusing because of it, in my opinion.